app/scripts/models/auth_brokers/mixins/channel.js

```diff
@@ -17,11 +17,48 @@ define([
   'underscore',
   'lib/promise'
 ], function (_, p) {
+  function doIt(strategy, channel, message, data) {
+    var boundStrategy = strategy.bind(channel);
+    // new channels are already set up to return promises. If so,
+    // no need to denodeify.
+    if (strategy.length === 3) {
+      return p.denodeify(boundStrategy)(message, data);
+    }
+
+    return boundStrategy(message, data);
+  }
+
   var ChannelMixin = {
+    /**
```

Comment:
The long term goal is to get rid of this mixin completely and have all the channels converted to return promises.